com.unity.cinemachine/Editor/Upgrader/CinemachineUpgrader.cs

```diff
@@ -642,24 +644,78 @@ void Initialize(List<PlayableDirector> playableDirectors)
                         }
                     }
                 }
+                
+                // local function
+                static void LoadApiMapping()
+                {
+                    if (s_ApiUpgradeMaps != null)
+                        return;
+
+                    var filename = ScriptableObjectUtility.CinemachineRelativeInstallPath + 
+                        "/Editor/EditorResources/ApiUpgradeMapping.json";
+                    var json = System.IO.File.ReadAllText(filename);
+                    s_ApiUpgradeMaps = Newtonsoft.Json.JsonConvert
+                        .DeserializeObject<Dictionary<Type, Dictionary<string, Tuple<string, Type>>>>(json);
+                }
             }
+
+            static readonly Dictionary<Type, Type> k_ClassUpgradeMaps = new()
+            {
+                {typeof(CinemachineVirtualCamera), typeof(CmCamera)},
+                {typeof(CinemachineFreeLook), typeof(CmCamera)},
+                {typeof(CinemachineComposer), typeof(CinemachineRotationComposer)},
+                {typeof(CinemachineGroupComposer), typeof(CinemachineRotationComposer)},
+                {typeof(CinemachineTransposer), typeof(CinemachineFollow)},
+                {typeof(CinemachineFramingTransposer), typeof(CinemachinePositionComposer)},
+                {typeof(CinemachinePOV), typeof(CinemachinePanTilt)},
+                {typeof(CinemachineOrbitalTransposer), typeof(CinemachineOrbitalFollow)},
+                {typeof(CinemachineTrackedDolly), typeof(CinemachineSplineDolly)},
+                {typeof(CinemachinePath), typeof(SplineContainer)},
+                {typeof(CinemachineSmoothPath), typeof(SplineContainer)},
+                {typeof(CinemachineDollyCart), typeof(CinemachineSplineCart)},
+#if CINEMACHINE_UNITY_INPUTSYSTEM
+                {typeof(CinemachineInputProvider), typeof(InputAxisController)},
+#endif
+            };
+
+            static Dictionary<Type, Dictionary<string, Tuple<string, Type>>> s_ApiUpgradeMaps;
             
             static void ProcessAnimationClip(AnimationClip animationClip)
             {
                 var existingEditorBindings = AnimationUtility.GetCurveBindings(animationClip);
                 foreach (var previousBinding in existingEditorBindings)
                 {
                     var newBinding = previousBinding;
```

Comment:
One way to improve abstaction is to move this logic out of here and into an UpgradeAnimationClip function in the cm2tocm3 module.